src/Ledger.Net/LedgerManager.cs

```diff
@@ -156,31 +156,57 @@ protected byte[] GetResponseDataPacket(byte[] data, int packetIndex, ref int rem
         #endregion
 
         #region Public Methods
-        public async Task<string> GetAddressAsync(uint coinNumber, uint account, bool isChange, uint index, bool showDisplay, AddressType addressType, bool isSegwit)
+        public async Task<string> GetAddressAsync(AddressType addressType, uint addressIndex)
         {
-            var indices = new[] { ((isSegwit ? (uint)49 : 44) | Constants.HARDENING_CONSTANT) >> 0, (coinNumber | Constants.HARDENING_CONSTANT) >> 0, (0 | Constants.HARDENING_CONSTANT) >> (int)account, isChange ? 1 : (uint)0, index };
-
-            byte[] addressIndicesData;
+            GetPublicKeyResponseBase response;
+            if (addressType == AddressType.Ethereum)
+            {
+                response = await SendRequestAsync<EthereumAppGetPublicKeyResponse, EthereumAppGetPublicKeyRequest>(new EthereumAppGetPublicKeyRequest(
+                    addressIndex,
+                    false,
+                    false));
+            }
+            else
+            {
+                response = await SendRequestAsync<BitcoinAppGetPublicKeyResponse, BitcoinAppGetPublicKeyRequest>(new BitcoinAppGetPublicKeyRequest(
+                    addressIndex,
+                    false));
+            }
 
-            using (var memoryStream = new MemoryStream())
+            if (!response.IsSuccess)
             {
-                memoryStream.WriteByte((byte)indices.Length);
-                for (var i = 0; i < indices.Length; i++)
-                {
-                    var data = indices[i].ToBytes();
-                    memoryStream.Write(data, 0, data.Length);
-                }
-                addressIndicesData = memoryStream.ToArray();
+                throw new Exception(response.StatusMessage);
             }
 
+            return response.Address;
+        }
+
+        public async Task<string> GetAddressAsync(
```

Comment:
This is a controversial issue, and I may be in the minority here, but I value vertical terseness as much as horizontal terseness. I like method signatures on one line. Call me crazy, but it's a personal preference, and I turn on line wrapping so this isn't necessary.